middlewares/osio/cache.go

```diff
@@ -41,15 +41,7 @@ func (r *ResolverPromise) Get() (interface{}, error) {
 		return r.value, r.err
 	}
 
-	wg := sync.WaitGroup{}
-	wg.Add(1)
-	go func(wg *sync.WaitGroup, resolver Resolver) {
-		r.value, r.err = resolver()
-		wg.Done()
-
-	}(&wg, r.resolver)
-	wg.Wait()
-
+	r.value, r.err = r.resolver()
```

Comment:
@aslakknutsen Call to resovler() done in same go routine.  Here, caller go routine (i.e. Get() executor go routine) always block for resolver() to complete.  So rather calling resolver() in different go routine.  Now change is made to call in same caller go routine.